### PR TITLE
Switch to new style for order cancelation methods

### PR DESCRIPTION
**Pull request by ashkan18**
# Change
We've switched most of our actions to new style of services where we have `OrderService` have the business logic of what happens during in each action and when needed depending on the logic, use other helpers/processors to handle the logic.

This follows the same pattern and is the last piece left to fully move all of our actions to this style and be more consistent:
https://artsyproduct.atlassian.net/browse/PURCHASE-902

# Extra
As part of this added few new `statsd` counters, because why not?

### CODE REVIEW

File: app/admin/orders.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@
   end
 
   member_action :refund, method: :post do
-    OrderCancellationService.new(resource).refund!
+    OrderServoce.refund!(resource)
```
Comment (sweir27) [inline app/admin/orders.rb]:
```suggestion
OrderService.refund!(resource)
```

Comment (ashkan18) [reply app/admin/orders.rb]: clearly there is not much test around admin... thanks for 👀

File: app/services/order_service.rb [hunk 2]
```diff
@@ -123,5 +123,42 @@ def self.confirm_fulfillment!(order, user_id)
...
+    order_cancelation_processor.cancel_payment if order.mode == Order::BUY
+    order_cancelation_processor.queue_undeduct_inventory_jobs if order.mode == Order::BUY
+    order_cancelation_processor.notify
+    Exchange.dogstatsd.increment 'order.reject'
+  end
+
+  def self.refund!(order)
+    order.refund!
+    order_cancelation_processor = OrderCancellationProcessor.new(order)
+    order_cancelation_processor.refund_payment
```
Comment (sweir27) [inline app/services/order_service.rb]: If someone mistakenly tries to `refund` on a BUY order, will this error in the way we expect?

File: lib/order_cancellation_processor.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+class OrderCancellationProcessor
```
Comment (sweir27) [inline lib/order_cancellation_processor.rb]: For consistency, can we name this `OrderCancelationProcessor`? 🙏

File: app/services/order_service.rb (hunk 2 above)
Comment (ashkan18) [reply app/services/order_service.rb]: in the admin, we only allow `approved` and `fulfilled` orders to be refunded. Also in the model we can only go to refund state from those two states so we should be safe
https://github.com/artsy/exchange/blob/master/app/models/order.rb#L262

File: app/services/order_service.rb [hunk 4]
```diff
@@ -123,5 +123,42 @@ def self.confirm_fulfillment!(order, user_id)
...
+  end
+
+  def self.refund!(order)
+    order.refund!
+    order_cancelation_processor = OrderCancelationProcessor.new(order)
+    order_cancelation_processor.refund_payment
+    order_cancelation_processor.queue_undeduct_inventory_jobs
+    order_cancelation_processor.notify
+    Exchange.dogstatsd.increment 'order.refund'
+    Exchange.dogstatsd.count('order.money_refunded', order.buyer_total_cents)
```
Comment (iskounen) [inline app/services/order_service.rb]: Should we have separate order.money_refunded buckets for different currencies?

Comment (ashkan18) [reply app/services/order_service.rb]: good call! i don't think anyone is looking at this at the moment but let me update to include currency in bucket name.

Comment (ashkan18) [reply app/services/order_service.rb]: @iskounen made that change, take a look when have a chance

**Comment by ashkan18** [comment]
going to merge this to have it on staging for few hours and do some testing.

**Pull request merged**